### PR TITLE
Fix redux

### PR DESCRIPTION
**Pull request by hueneburg**
Fix redux regarding state in url.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -302,3 +302,17 @@ gradle-app.setting
 
 
 # End of https://www.gitignore.io/api/eclipse,gradle,intellij+iml,node,grunt,typings,visualstudiocode,windows,linux,macos,vim,jabref
+
```
Comment (kleinech) [inline .gitignore]: Revert changes and remove `.js` and `.js.map` files

File: src/app/app.component.ts [hunk 2]
```diff
@@ -1,9 +1,16 @@
-import { Component, ViewEncapsulation } from '@angular/core';
+import { Component, ViewChild, ViewEncapsulation } from '@angular/core';
```
Comment (kleinech) [inline src/app/app.component.ts]: Can ViewChild be removed?

File: src/app/components/comparison/comparison.template.html [hunk 3]
```diff
@@ -71,7 +71,7 @@ <h1>{{configurationService.configuration?.title}}
     <div class="grid-content-end">
         <pcard heading="References" *ngIf="configurationService.configuration.citation.size != 0">
             <referencestable [changeNum]="configurationService.configuration.citation.length"
-                             [citationMap]="configurationService.configuration.citation"></referencestable>
+                             [citationMap]="configurationService.configuration.citation" [prefix]=""></referencestable>
```
Comment (kleinech) [inline src/app/components/comparison/comparison.template.html]: Remove empty prefix (use default for prefix)

File: src/app/components/polymer/paper-dialog/paper-dialog.component.ts [hunk 4]
```diff
@@ -13,7 +13,8 @@ export class PaperDialogComponent {
 
     @HostListener('click', ['$event.target']) onClick(target) {
         if (target.localName === 'pdialog') {
-            this.close();
+            this.opened = false;
```
Comment (kleinech) [inline src/app/components/polymer/paper-dialog/paper-dialog.component.ts]: Explain changes

File: src/app/redux/uc.reducers.ts [hunk 5]
```diff
@@ -169,28 +208,28 @@ function changeOrder(state: IUCAppState, action: UCTableOrderAction): IUCAppStat
...
     }
-    putStateIntoURL(state);
     return state;
 }
 
 function initSettings(state: IUCAppState): IUCAppState {
     // Set elements settings
     const elementNames: Array<string> = [];
     const elementsEnabled: Array<boolean> = [];
-    DataService.data.forEach(value => {
```
Comment (kleinech) [inline src/app/redux/uc.reducers.ts]: DataService.data.forEach((value, index) => {...

https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach

File: src/app/redux/uc.reducers.ts [hunk 6]
```diff
@@ -339,10 +404,13 @@ function filterElements(state: IUCAppState, criterias: Map<string, Criteria> = n
...
             const criteria = state.criterias.get(field);
+            if (isNullOrUndefined(criteria)) {
+                continue;
+            }
             if (criteria.rangeSearch) {
-                if (state.currentSearch.get(field).length > 0) {
-                    const queries = (state.currentSearch.get(field)[0] || '').trim().replace(' ', '')
-                        .replace(/,.*[a-zA-Z].*|.*[a-zA-Z].*,|.*[a-zA-Z].*/g, '').split(',');
+                if (state.currentSearch.get(field).size > 0) {
+                    const queries = (state.currentSearch.get(field).values().next().value || '').trim().replace(' ', '')
```
Comment (kleinech) [inline src/app/redux/uc.reducers.ts]: Please write a comment: Query...

**Comment by kleinech** [comment]
Please fix codeclimate issues (e.g.: similar code, @Hostlistener, debugger;)

and merge changes to master into this branch

File: src/app/redux/uc.reducers.ts (hunk 5 above)
Comment (hueneburg) [reply src/app/redux/uc.reducers.ts]: also solved, github just doesn't notice for some reason...

**Pull request merged**